keps/sig-scheduling/3902-decoupled-taint-manager/README.md

```diff
@@ -0,0 +1,271 @@
+
+#### GA
+
+* Fix all reported bugs if any.
+
+### Upgrade / Downgrade Strategy
+
+A feature gate `SeparateTaintManager`  enables and disables the new feature. When the feature is turned on, a user can use `kube-controller-manager`'s flag to disable the default `TaintManager`.
+
+
+### Version Skew Strategy
+
+## Production Readiness Review Questionnaire
+
+### Feature Enablement and Rollback
+
+*This section must be completed when targeting alpha to a release.*
+
+* **How can this feature be enabled / disabled in a live cluster?**
+    * Enabled:  turn on the new feature gate `SeparateTaintManager`
+    * Disabled: turn off the new feature gate `SeparateTaintManager`
+* **Does enabling the feature change any default behavior?**
+    * No, the default node-life-cycle controller and taint-manager behavior will stay the same.
+* **Can the feature be disabled once it has been enabled (i.e. can we roll back the enablement)?**
+    * Yes, we will need to rack back `kube-controller-manager` and `node-lifecycle-controller`. 
+* **What happens if we re-enable the feature if it was previously rolled back?**
+    * Use the feature gate `SeparateTaintManager` to enable or disable the feature.
+* **Are there any tests for feature enablement/disablement?**
+    * Planned for Beta release
+
+### Rollout, Upgrade and Rollback Planning
+
+*This section must be completed when targeting beta graduation to a release.*
+
+* **How can a rollout or rollback fail? Can it impact already running workloads?**
+  A feature gate `SeparateTaintManager` controls whether to use the split `TaintManager` or the old combined `NodeLifecycleController`.
+
+* **What specific metrics should inform a rollback?**
+  N/A
+
+* **Were upgrade and rollback tested? Was the upgrade→downgrade→upgrade path tested?**
+  They will be covered in unit and e2e tests.
+
+* **Is the rollout accompanied by any deprecations and/or removals of features, APIs, fields of API types, flags, etc.?**
+  No
+
+### Monitoring Requirements
+
+*This section must be completed when targeting beta graduation to a release.*
+
+* **How can an operator determine if the feature is in use by workloads?**
+    * Disable the default taint-manager
+    * Create custom taint-based pod eviction
+* **How can someone using this feature know that it is working for their instance?**
+    * Node taints and taint-based pod eviction work as usual. Admins and users should not see any different behavior if everything works as planned.
+* **What are the reasonable SLOs (Service Level Objectives) for the enhancement?**
+    * Taint-based pod eviction latency (e.g., P99) should be less than x seconds.
+* **What are the SLIs (Service Level Indicators) an operator can use to determine the health of the service?**
+    * Taint-based pod eviction latency
+* **Are there any missing metrics that would be useful to have to improve observability of this feature?**
+    * Taint-based pod eviction latency 
```

Comment:
Can you point out the exact metric name?

Reply:
Will change it to `No` as this change should have minimal performance impact on the pod eviction behavior.